src/java/org/apache/cassandra/journal/Params.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.journal;
+
+public interface Params
+{
+    enum FlushMode { BATCH, GROUP, PERIODIC }
+
+    enum FailurePolicy { STOP, STOP_JOURNAL, IGNORE, DIE }
+
+    /**
+     * @return maximum segment size
+     */
+    int segmentSize();
+
+    /**
+     * @return this journal's {@link FailurePolicy}
+     */
+    FailurePolicy failurePolicy();
+
+    /**
+     * @return journal flush (sync) mode
+     */
+    FlushMode flushMode();
+
+    /**
+     * @return milliseconds between journal flushes
+     */
+    int flushPeriod();
+
+    /**
+     * @return milliseconds to block writes for while waiting for a slow disk flush to complete
+     *         when in {@link FlushMode#PERIODIC} mode
+     */
+    int periodicFlushLagBlock();
+
+    /**
+     * @return user provided version to use for key and value serialization
+     */
+    int userVersion();
```

Comment:
my last comment was to expose this via yaml, but we wouldn't want this as an `int` there, we would want an `enum` (allow opt-into new versions)...  guess we could use `@JsonIgnore` to hid it...

guess `Params` shouldn't be yaml, but likely `DEFAULT` would, where it takes an enum and maps that to this `int`

Reply:
If this should always be `MessagingService`'s version for all users, then we can use `org.apache.cassandra.net.MessagingService.Version`...  I added this to get access to versions w/o loading `MessagingService` for tests, but could use in this context as well.

Reply:
Intended to not be tied to `MessagingService` version and be allowed to evolve (or not evolve) somewhat independently from the rest of the system. We currently couple things too tightly to MS for little good reason.

Reply:
> We currently couple things too tightly to MS for little good reason.

yeah, this has been annoying for other unrelated patches, as touching 1 section that can evolve independently must push everyone forward...

> Intended to not be tied to MessagingService version and be allowed to evolve (or not evolve) somewhat independently from the rest of the system.

since this API isn't for `cassandra.yml` and is for a pluggable config for higher level, this is fine.  It would be good to expose `accord_journal` in yaml with a `enum` version (don't force `CURRENT`, great for downgrades)